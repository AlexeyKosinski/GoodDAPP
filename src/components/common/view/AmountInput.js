// @flow
import React, { useState } from 'react'
import { Keyboard, TouchableWithoutFeedback, View } from 'react-native'
import { isMobile } from 'mobile-device-detect'
import InputGoodDollar from '../form/InputGoodDollar'
import { withStyles } from '../../../lib/styles'
import Text from '../view/Text'
import NumPadKeyboard from './NumPadKeyboard'

type AmountInputProps = {
  amount: string,
  handleAmountChange: Function,
  styles: any,
}

const AmountInput = ({ amount, handleAmountChange, styles, error }: AmountInputProps) => {
  const [caretPosition, setCaretPosition] = useState({ start: 0, end: 0 })

  return (
    <View style={styles.wrapper}>
      <View>
        <TouchableWithoutFeedback
          onPress={() => (isMobile ? Keyboard.dismiss() : null)}
          accessible={false}
          style={styles.section}
        >
          <InputGoodDollar
            style={error ? styles.errorInput : {}}
            disabled={isMobile}
            autoFocus
            amount={amount}
            onChangeAmount={handleAmountChange}
            onSelectionChange={setCaretPosition}
          />
        </TouchableWithoutFeedback>
        <Text style={styles.errorText}>{error}</Text>
      </View>
      <NumPadKeyboard
        amount={amount}
        onPress={handleAmountChange}
        caretPosition={caretPosition}
        updateCaretPosition={setCaretPosition}
      />
    </View>
  )
}

const mapPropsToStyles = ({ theme }) => {
  return {
    wrapper: {
      width: '100%',
      display: 'flex',
      justifyContent: 'space-between',
      flex: 1,
    },
    errorText: {
      color: theme.colors.error,
      marginTop: theme.sizes.default,
    },
    errorInput: {
      color: theme.colors.error,
      borderBottomColor: theme.colors.error,
    },
  }
}

export default withStyles(mapPropsToStyles)(AmountInput)
