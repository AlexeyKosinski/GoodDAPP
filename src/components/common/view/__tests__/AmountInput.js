import React from 'react'
import renderer from 'react-test-renderer'
import AmountInput from '../AmountInput'
import { withThemeProvider } from '../../../../__tests__/__util__'

describe('AmountInput', () => {
  const WrappedAmountInput = withThemeProvider(AmountInput)

  let amount = '14'
  const handleAmountChange = value => {
    amount = value
  }

  it('renders without errors', () => {
    const tree = renderer.create(<WrappedAmountInput handleAmountChange={handleAmountChange} amount={amount} />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<WrappedAmountInput handleAmountChange={handleAmountChange} amount={amount} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
