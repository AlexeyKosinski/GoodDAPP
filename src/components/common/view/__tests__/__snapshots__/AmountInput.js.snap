// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountInput matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitBoxFlex": 1,
        "WebkitBoxPack": "justify",
        "WebkitFlexBasis": "0%",
        "WebkitFlexGrow": 1,
        "WebkitFlexShrink": 1,
        "WebkitJustifyContent": "space-between",
        "display": "flex",
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
        "msFlexNegative": 1,
        "msFlexPack": "justify",
        "msFlexPositive": 1,
        "msFlexPreferredSize": "0%",
        "width": "100%",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
    >
      <input
        autoCapitalize="sentences"
        autoComplete="on"
        autoCorrect="on"
        autoFocus={true}
        className="css-textinput-1cwyjr8"
        dir="auto"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onSelect={[Function]}
        onStartShouldSetResponder={[Function]}
        placeholder="0 G$"
        readOnly={false}
        spellCheck={true}
        style={
          Object {
            "borderBottomColor": "rgba(66,69,74,1.00)",
            "borderBottomStyle": "solid",
            "borderBottomWidth": "1px",
            "color": "rgba(66,69,74,1.00)",
            "cursor": "pointer",
            "fontFamily": "RobotoSlab-Regular",
            "fontSize": "32px",
            "fontWeight": "bold",
            "letterSpacing": "1.2px",
            "msTouchAction": "manipulation",
            "paddingBottom": "4px",
            "paddingLeft": "4px",
            "paddingRight": "4px",
            "paddingTop": "4px",
            "textAlign": "center",
            "touchAction": "manipulation",
          }
        }
        styles={
          Object {
            "input": Object {
              "borderBottomColor": "#42454A",
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "color": "#42454A",
              "fontFamily": "RobotoSlab-Regular",
              "fontSize": 18,
              "padding": 4,
              "textAlign": "center",
            },
          }
        }
        tabIndex="-1"
        theme={
          Object {
            "backdrop": Object {
              "backgroundColor": "rgba(255, 0, 0, 0.8)",
            },
            "borders": Object {
              "defaultBorderColor": "#c9c8c9",
            },
            "colors": Object {
              "accent": "#03dac4",
              "backdrop": "rgba(0, 0, 0, 0.5)",
              "background": "#f6f6f6",
              "blue": "#006EA0",
              "darkBlue": "#0C263D",
              "darkGray": "#42454A",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "error": "#FA6C77",
              "gray": "#555",
              "gray50Percent": "#CBCBCB",
              "gray80Percent": "#A3A3A3",
              "green": "#00C3AE",
              "lightGray": "#EEE",
              "lighterGray": "#A3A3A3",
              "notification": "#f50057",
              "orange": "#F8AF40",
              "placeholder": "#CBCBCB",
              "primary": "#00AFFF",
              "purple": "#9F6A9D",
              "red": "#FA6C77",
              "surface": "#ffffff",
              "text": "#222",
            },
            "dark": false,
            "feedItems": Object {
              "borderRadius": 8,
              "height": 84,
              "itemBackgroundColor": "#fff",
            },
            "fontStyle": Object {
              "color": "#555",
              "fontSize": 18,
              "textAlign": "center",
            },
            "fonts": Object {
              "bold": "Roboto-Bold",
              "light": "Roboto-Light, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              "medium": "Roboto-Medium",
              "regular": "Roboto",
              "slab": "RobotoSlab-Regular",
              "slabBold": "RobotoSlab-Bold",
              "thin": "Roboto-Thin, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            },
            "modals": Object {
              "backgroundColor": "#fff",
              "borderLeftWidth": 10,
              "borderRadius": 5,
              "contentPadding": 16,
              "jaggedEdgeSize": 15,
              "overlayBackgroundColor": "rgba(255, 255, 255, 0.8)",
              "overlayHorizontalPadding": 20,
              "overlayVerticalPadding": 30,
            },
            "paddings": Object {
              "defaultMargin": 8,
              "mainContainerPadding": 8,
            },
            "roundness": 4,
            "sizes": Object {
              "borderRadius": 5,
              "default": 8,
              "defaultDouble": 16,
              "defaultHalf": 4,
            },
          }
        }
        type="text"
        value="14"
      />
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(250,108,119,1.00)",
            "direction": "ltr",
            "fontFamily": "Roboto",
            "fontSize": "16px",
            "fontWeight": "normal",
            "lineHeight": "22px",
            "marginTop": "8px",
            "textAlign": "center",
            "textDecoration": "none",
            "textTransform": "none",
          }
        }
      />
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "display": "flex",
          "marginBottom": "8px",
          "marginTop": "16px",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitAlignItems": "center",
            "WebkitBoxAlign": "center",
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "horizontal",
            "WebkitBoxPack": "justify",
            "WebkitFlexDirection": "row",
            "WebkitJustifyContent": "space-between",
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "msFlexAlign": "center",
            "msFlexDirection": "row",
            "msFlexPack": "justify",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            1
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            2
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            3
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitAlignItems": "center",
            "WebkitBoxAlign": "center",
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "horizontal",
            "WebkitBoxPack": "justify",
            "WebkitFlexDirection": "row",
            "WebkitJustifyContent": "space-between",
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "msFlexAlign": "center",
            "msFlexDirection": "row",
            "msFlexPack": "justify",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            4
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            5
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            6
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitAlignItems": "center",
            "WebkitBoxAlign": "center",
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "horizontal",
            "WebkitBoxPack": "justify",
            "WebkitFlexDirection": "row",
            "WebkitJustifyContent": "space-between",
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "msFlexAlign": "center",
            "msFlexDirection": "row",
            "msFlexPack": "justify",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            7
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            8
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            9
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitAlignItems": "center",
            "WebkitBoxAlign": "center",
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "horizontal",
            "WebkitBoxPack": "justify",
            "WebkitFlexDirection": "row",
            "WebkitJustifyContent": "space-between",
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "msFlexAlign": "center",
            "msFlexDirection": "row",
            "msFlexPack": "justify",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            .
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(66,69,74,1.00)",
                "direction": "ltr",
                "fontFamily": "RobotoSlab-Bold",
                "fontSize": "20px",
                "fontWeight": "700",
                "lineHeight": "22px",
                "textAlign": "center",
                "textDecoration": "none",
                "textTransform": "none",
              }
            }
          >
            0
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxFlex": 1,
              "WebkitFlexBasis": "0%",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
              "msFlexAlign": "center",
              "msFlexNegative": 1,
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0%",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "20px",
              "paddingLeft": "20px",
              "paddingRight": "20px",
              "paddingTop": "20px",
              "touchAction": "manipulation",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "backgroundImage": "url(backKeyboardButton.png)",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
                "cursor": "pointer",
                "height": "30px",
                "width": "40px",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
